crates/storage/src/connection/event.rs

```diff
@@ -48,94 +42,60 @@ pub enum EventFilterError {
     TooManyMatches,
 }
 
+impl From<rusqlite::Error> for EventFilterError {
+    fn from(error: rusqlite::Error) -> Self {
+        Self::Internal(error.into())
+    }
+}
+
+impl From<crate::bloom::CacheError> for EventFilterError {
+    fn from(error: crate::bloom::CacheError) -> Self {
+        Self::Internal(error.into())
+    }
+}
+
 #[derive(Clone, Debug, PartialEq, Eq)]
 pub struct PageOfEvents {
     pub events: Vec<EmittedEvent>,
     pub is_last_page: bool,
 }
 
-pub trait KeyFilter {
-    fn count(&self, tx: &Transaction<'_>) -> anyhow::Result<Option<usize>>;
-    fn apply(&self, strategy: QueryStrategy) -> Option<KeyFilterResult<'_>>;
-}
-
 #[derive(Debug, PartialEq)]
 pub struct KeyFilterResult<'a> {
     pub base_query: &'static str,
     pub where_statement: &'static str,
     pub param: (&'static str, rusqlite::types::ToSqlOutput<'a>),
 }
 
-pub(super) fn insert_events(
+pub(super) fn insert_block_events<'a>(
```

Comment:
Should this also insert into the cache? Making the assumption that new block data is more relevant generally.

Reply:
Maybe -- but I don't think it matters much. It's a single load -- _if_ there are queries for this block.